h2o-algos/src/main/java/hex/svd/SVD.java

```diff
@@ -701,9 +706,12 @@ public void computeImpl() {
           }
 
           for (int k = 1; k < _parms._nv; k++) {  // loop through for each eigenvalue/eigenvector...
-            if (stop_requested()) break;
-            if (_matrixRankReached) { // number of eigenvalues found is less than _nv
-              int newk = k-1;
+            if (_matrixRankReached || stop_requested()) { // number of eigenvalues found is less than _nv
+              if (stop_requested()) {
+                _matrixRankReached = true;
```

Comment:
This is needed because:

1. Some datasets are rank deficient.  Hence, a dataset is of rank 10 say and a user is asking for k = 12 which is 2 more than what the dataset's true rank is.  The PCA can detect this and will therefore set the k to be 10 instead of 12.  In this case, a warning will be sent back to the client to let them know this.
2. Due to max_runtime_secs restriction, we may not have time to find the number of eigenvalues/eigenvectors specified by the user in parms._nv.  Hence, in this case again, we will need to reduce the parameter to be the actual number of eigenvalue/eigenvector we found instead of the ones asked by the user.